step/src/main/java/servlets/MatchingServlet.java

```diff
@@ -23,9 +29,25 @@
 @WebServlet("/matching")
 public class MatchingServlet extends HttpServlet {
 
+  public static String REQUEST_TYPE = "requestType";
+  public static String REQUEST_TYPE_MATCH = "match-request";
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // TODO: implement
   }
 
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String requestType = request.getParameter(REQUEST_TYPE);
+    if(requestType.equals(REQUEST_TYPE_MATCH)) {
+      UserService userService = UserServiceFactory.getUserService();
+      com.google.appengine.api.users.User userServiceUser = userService.getCurrentUser();
```

Comment:
Can you import the API as a library above the class name?

Reply:
It's already imported but I chose to use the full class name to reduce confusion (for the compiler and programmer) about which User object is the API's and which is ours.